[python3] Build error

Package: python3:x64-linux -> 3.11.4#4

**Host Environment**

- Host: x64-linux
- Compiler: GNU 10.2.1
-    vcpkg-tool version: 2023-08-09-9990a4c9026811a312cb2af78bf77f3d9d288416
    vcpkg-scripts version: 436a061 2023-08-25 (3 days ago)

**To Reproduce**

`vcpkg install python3`
**Failure logs**

```
-- Downloading https://github.com/python/cpython/archive/v3.11.4.tar.gz -> python-cpython-v3.11.4.tar.gz...
-- Extracting source /vcpkg/downloads/python-cpython-v3.11.4.tar.gz
-- Applying patch 0001-only-build-required-projects.patch
-- Applying patch 0003-use-vcpkg-zlib.patch
-- Applying patch 0004-devendor-external-dependencies.patch
-- Applying patch 0005-dont-copy-vcruntime.patch
-- Applying patch 0008-python.pc.patch
-- Applying patch 0010-dont-skip-rpath.patch
-- Applying patch 0012-force-disable-curses.patch
-- Applying patch 0014-fix-get-python-inc-output.patch
-- Applying patch 0015-dont-use-WINDOWS-def.patch
-- Applying patch 0002-static-library.patch
-- Applying patch 0011-gcc-ldflags-fix.patch
-- Using source at /vcpkg/buildtrees/python3/src/v3.11.4-101b2af427.clean
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
-- Generating configure for x64-linux
-- Finished generating configure for x64-linux
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Installing x64-linux-dbg
-- Building x64-linux-rel
-- Installing x64-linux-rel
-- Fixing pkgconfig file: /vcpkg/packages/python3_x64-linux/lib/pkgconfig/python-3.11-embed.pc
-- Fixing pkgconfig file: /vcpkg/packages/python3_x64-linux/lib/pkgconfig/python-3.11.pc
-- Fixing pkgconfig file: /vcpkg/packages/python3_x64-linux/debug/lib/pkgconfig/python-3.11-embed.pc
-- Fixing pkgconfig file: /vcpkg/packages/python3_x64-linux/debug/lib/pkgconfig/python-3.11.pc
CMake Error at ports/python3/portfile.cmake:300 (message):
  There should be no modules with "_failed" suffix:


  /vcpkg/packages/python3_x64-linux/debug/lib/python3.11/lib-dynload/_ctypes.cpython-311d-x86_64-linux-gnu_failed.so



  /vcpkg/packages/python3_x64-linux/lib/python3.11/lib-dynload/_ctypes.cpython-311-x86_64-linux-gnu_failed.so
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)



```

Reproducable by using the container image `quay.io/pypa/manylinux2014_x86_64` (which is a common container to use for building native Python libraries that have to work on many different flavours of Linux).

--- a/ports/libffi/portfile.cmake
+++ b/ports/libffi/portfile.cmake
@@ -48,6 +48,7 @@ vcpkg_configure_make(
     OPTIONS
         --enable-portable-binary
         --disable-docs
+        --disable-multi-os-directory
         ${options}
 )
 

--- a/ports/libffi/vcpkg.json
+++ b/ports/libffi/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libffi",
   "version": "3.4.4",
-  "port-version": 3,
+  "port-version": 4,
   "description": "Portable, high level programming interface to various calling conventions",
   "homepage": "https://github.com/libffi/libffi",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4094,7 +4094,7 @@
     },
     "libffi": {
       "baseline": "3.4.4",
-      "port-version": 3
+      "port-version": 4
     },
     "libfido2": {
       "baseline": "1.13.0",

--- a/versions/l-/libffi.json
+++ b/versions/l-/libffi.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d189744d6e4f29ab0a2f88f3c75f01d0c3719618",
+      "version": "3.4.4",
+      "port-version": 4
+    },
+    {
       "git-tree": "5229da2e57e35c0ab7d043b27c19a29355e0a2c2",
       "version": "3.4.4",
       "port-version": 3

